.github/workflows/validate-demo-metadata.yml

```
@@ -1,37 +1,151 @@
+          
+          for metadata_file_name in $input_metadata_file_names
+          do
+            if [[ "$metadata_file_name" =~ "../demonstrations" ]]; then
+              formatted_metadata_file_names+=("$metadata_file_name")
+            elif [[ "$metadata_file_name" =~ "demonstrations" ]]; then
+              formatted_metadata_file_names+=("../$metadata_file_name")
+            else
+              formatted_metadata_file_names+=("../demonstrations/$metadata_file_name")
+            fi
+          done
+          
+          echo "formatted_file_names=${formatted_metadata_file_names[@]}" >> $GITHUB_OUTPUT
+
+    outputs:
+      metadata_files: ${{ steps.format_metadata_files.outputs.formatted_file_names }}
+
+  validate-metadata-schema:
+    runs-on: ubuntu-22.04
+
+    if: inputs.skip_metadata_schema_validation == false
+
+    needs:
+      - generate-metadata-file-list
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 1
+          ref: ${{ inputs.branch }}
 
       - name: Setup Python
         uses: actions/setup-python@v4
         with:
           python-version: 3.9
 
-      - name: Install Python Dependencies
+      - name: Install Poetry
+        id: poetry
+        env:
+          POETRY_HOME: /tmp/poetry
         run: |
-          pip install poetry
-          poetry install --only metadata-validation
+          curl -sSL https://install.python-poetry.org -o install-poetry.py
+          python3 install-poetry.py --version 1.5.1
+          echo "bin=${{ env.POETRY_HOME }}/bin/poetry" >> $GITHUB_OUTPUT
 
-      - name: Validate Metadata
+      - name: Install Metadata Schema Validation Dependencis
+        run: ${{ steps.poetry.outputs.bin }} install --only metadata-validation
+
+      - name: Validate Metadata Schema
+        env:
+          METADATA_FILE_LIST: ${{ needs.generate-metadata-file-list.outputs.metadata_files }}
         run: |
           cd metadata_schemas
-          files=`find ../demonstrations -name "*.metadata.json" -not -path "*/demonstrations_categories.metadata.json" | tr '\n' ' '`
-          poetry run check-jsonschema -v --traceback-mode full --schemafile demo.metadata.schema.0.1.0.json $(echo $files)
+          ${{ steps.poetry.outputs.bin }} run check-jsonschema -v --traceback-mode full --schemafile demo.metadata.schema.0.1.0.json $METADATA_FILE_LIST
+
+  validate-metadata-preview-images:
+    runs-on: ubuntu-22.04
+
+    if: inputs.skip_metadata_preview_image_value_validation == false
+
+    needs:
+      - generate-metadata-file-list
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 1
+          ref: ${{ inputs.branch }}
+
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.9
+
+      - name: Install QML Pipeline Utils
+        run: |
+          cd .github/workflows/qml_pipeline_utils
+          pip install .
```

Can we use this instead?
```suggestion
          pip install .github/workflows/qml_pipeline_utils
```